src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx

```diff
@@ -49,10 +57,40 @@ export const Profile_PaymentOptionsScreen = ({}: PaymentOptionsProps) => {
   }
 
   useEffect(() => {
-    if (isFocused) {
-      refreshCards();
-    }
-  }, [isFocused]);
+    refreshCards();
+  }, []);
+
+  useEffect(() => {
+    const addPaymentMethodCallbackHandler = async ({url}: {url: string}) => {
+      if (
+        url.includes('response_code') &&
+        url.includes('recurring_payment_id')
+      ) {
+        // Timeout required to make loading spinner appear,
+        // see https://github.com/facebook/react-native/issues/37308
+        setTimeout(() => setIsLoading(true), 20);
```

Comment:
Could this also be solved by setting the initial state of `isLoading` as `false`, and to set it to `true` at start of loading, both here and in the `useEffect` on mount..

Reply:
No unfortunately, that doesn't work